sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollationExpressionSuite.scala

```diff
@@ -19,59 +19,75 @@ package org.apache.spark.sql.catalyst.expressions
 
 import org.apache.spark.{SparkException, SparkFunSuite}
 import org.apache.spark.sql.catalyst.util.CollationFactory
+import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 
 class CollationExpressionSuite extends SparkFunSuite with ExpressionEvalHelper {
   test("validate default collation") {
-    val collationId = CollationFactory.collationNameToId("UCS_BASIC")
-    assert(collationId == 0)
-    val collateExpr = Collate(Literal("abc"), "UCS_BASIC")
-    assert(collateExpr.dataType === StringType(collationId))
-    collateExpr.dataType.asInstanceOf[StringType].collationId == 0
-    checkEvaluation(collateExpr, "abc")
+    withSQLConf(SQLConf.COLLATION_ENABLED.key -> "true") {
```

Comment:
+1

Reply:
Another option is to enable the feature for testing purposes and disable it otherwise. I believe (not 100% sure) that this is doable if we define the config as:
```scala
val COLLATION_ENABLED =
    buildConf("spark.sql.collation.enabled")
      .doc("Collations feature is under development and its use should be done under this" +
        "feature flag.")
      .version("4.0.0")
      .booleanConf
      .createWithDefault(Utils.isTesting)
```